--- fcontext_ppc64_sysv_elf_gas.S.orig	2013-03-18 12:20:16.434349000 +0100
+++ fcontext_ppc64_sysv_elf_gas.S	2013-03-18 12:19:12.197494000 +0100
@@ -222,7 +222,8 @@
     mflr    %r4                 # load LR
     stw     %r4, 40(%r1)        # store LR on stack
     mr.     %r3, %r0            # context stack as arg to align_stack
-    bl      align_stack@plt     # call align_stack
+    bl      align_stack         # call align_stack
+    nop                         # nop is required by the linker
     mr.     %r0, %r3            # load result into R0
     lwz     %r4, 40(%r1)        # pop LR from stack
     mtlr    %r4                 # restore LR
@@ -246,5 +247,7 @@
 
 finish:
     li      %r3,  0             # set return value to zero
-    bl      _exit@plt           # exit application
+    bl      _exit               # exit application
+    nop                         # nop is required by the linker
+    .long   0                   # We should never get here, so SIGILL if we do
 .size .make_fcontext, .-.make_fcontext
