--- base_extender.h.orig	2013-04-03 17:11:52.049999000 +0200
+++ base_extender.h	2013-04-03 17:13:15.022112000 +0200
@@ -359,10 +359,10 @@
 		}
 		
 		Node source(const UEdge& edge) const {
-			return aNode(edge);
+			return this->aNode(edge);
 		}
 		Node target(const UEdge& edge) const {
-			return bNode(edge);
+			return this->bNode(edge);
 		}
 		
 		void firstInc(UEdge& edge, bool& dir, const Node& node) const {
